snuba/datasets/entities/outcomes.py

```diff
@@ -42,11 +47,12 @@ def __init__(self) -> None:
                     storage=materialized_storage,
                 ),
             ),
-            abstract_column_set=read_schema.get_columns(),
+            abstract_column_set=data_model,
             join_relationships={},
             writable_storage=writable_storage,
             validators=[EntityRequiredColumnValidator({"org_id"})],
             required_time_column="timestamp",
+            validate_data_model=ColumnValidationMode.WARN,
```

Comment:
Outcomes data model will still look the same after using the new `EntityDataModel` so we can start sending warnings to Sentry for invalid queries now